frame/bridge/ethereum/issuing/src/lib.rs

```diff
@@ -0,0 +1,423 @@
+		let mapped_address = dvm_ethereum::Module::<T>::do_call(factory_address, bytes)
+			.map_err(|e| -> &'static str { e.into() })?;
+		if mapped_address.len() != 32 {
+			return Err(Error::<T>::InvalidAddressLen.into());
+		}
+		Ok(H160::from_slice(&mapped_address.as_slice()[12..]))
+	}
+
+	pub fn finish_token_registered(
+		backing: EthereumAddress,
+		source: EthereumAddress,
+		target: EthereumAddress,
+	) -> DispatchResult {
+		let raw_event = RawEvent::TokenRegistered(0, backing, source, target);
+		let module_event: <T as Config>::Event = raw_event.clone().into();
+		let system_event: <T as frame_system::Config>::Event = module_event.into();
+		<BurnTokenEvents<T>>::append(system_event);
+		Self::deposit_event(raw_event);
+		T::EcdsaAuthorities::schedule_mmr_root(
+			(<frame_system::Pallet<T>>::block_number().saturated_into::<u32>() / 10 * 10 + 10)
+				.saturated_into(),
+		);
+		Ok(())
+	}
+
+	pub fn deposit_burn_token_event(
+		backing: EthereumAddress,
+		source: EthereumAddress,
+		recipient: EthereumAddress,
+		amount: U256,
+	) -> DispatchResult {
+		let mapped_address = Self::mapped_token_address(backing, source).map_err(|e| {
+			log::debug!("mapped token address error {:?} ", e);
+			e
+		})?;
+
+		let raw_event = RawEvent::BurnToken(
+			1,
+			backing,
+			recipient,
+			source,
+			mapped_address,
+			amount,
+		);
+		let module_event: <T as Config>::Event = raw_event.clone().into();
+		let system_event: <T as frame_system::Config>::Event = module_event.into();
+		<BurnTokenEvents<T>>::append(system_event);
+
+		Self::deposit_event(raw_event);
+		T::EcdsaAuthorities::schedule_mmr_root(
+			(<frame_system::Pallet<T>>::block_number().saturated_into::<u32>() / 10 * 10 + 10)
+				.saturated_into(),
+		);
+		Ok(())
+	}
+
+	pub fn verify_and_parse_proof(
+		log_event: EthEvent,
+		proof: EthereumReceiptProofThing<T>,
+	) -> Result<(EthereumTransactionIndex, EthLog), DispatchError> {
+		let tx_index = T::EthereumRelay::gen_receipt_index(&proof);
+		ensure!(
+			!VerifiedIssuingProof::contains_key(tx_index),
+			<Error<T>>::AssetAR
+		);
+		let verified_receipt = T::EthereumRelay::verify_receipt(&proof).map_err(|err| {
+			log::debug!("verify error {:?}", err);
+			<Error<T>>::ReceiptProofInv
+		})?;
+
+		let backing_address = EthereumBackingAddress::get();
+		let log_entry = verified_receipt
+			.logs
+			.into_iter()
+			.find(|x| x.address == backing_address && x.topics[0] == log_event.signature())
+			.ok_or(<Error<T>>::LogEntryNE)?;
+
+		let ethlog = Abi::parse_event(
+			log_entry.topics.into_iter().collect(),
+			log_entry.data.clone(),
+			log_event,
+		)
+		.map_err(|_| <Error<T>>::EthLogPF)?;
+
+		Ok((tx_index, ethlog))
+	}
+
+	pub fn transact_mapping_factory(input: Vec<u8>) -> DispatchResult {
+		let contract = MappingFactoryAddress::get();
+		let transaction = Self::unsigned_transaction(contract, input);
+		let result = dvm_ethereum::Module::<T>::do_transact(transaction, false).map_err(
```

Comment:
Suggest to remove the unsigned_transaction here, and using a new internal call `mock_transact(input: Vec<u8>)`